diff.py

```diff
@@ -2202,13 +2202,374 @@ def is_end_of_function(self, mnemonic: str, args: str) -> bool:
         return mnemonic == "ret"
 
 
+# pc-relative mov instructions
+# Examples:
+# "mov.l   150 <_bar+0x12>,r4     ! 154 <_main>"
+# "mov.w   266 <_main+0x112>,r4   ! bbaa"
+# "mova    190 <_main+0x3c>,r0"
+SH_POOL_PATTERN = (
+    r"(^.+mov\.?([lwa])\s+)([a-fA-F0-9]+)\s*<.+>(,r[0-9]+)(?:\s+!\s*([a-fA-F0-9]*))?"
+)
+
+# Normalized version of the normal pattern that uses pc-relative notation
+# Examples:
+# "mov.l   @(0x10,pc),r4 ! 150"
+# "mov.w   @(0x6e,pc),r4 ! 266"
+# "mova    @(0x22,pc),r0 ! 190"
+SH_POOL_PATTERN_NORM = (
+    r"(mov\.?[alw]\s+@\(0x[a-fA-F0-9]+,pc\).*,(r[0-9]|r1[0-5])+\s+!)\s+([a-fA-F0-9]+).*"
+)
+
+
 class AsmProcessorSH2(AsmProcessor):
+    @dataclass
+    class ImmEntry:
+        value: int
+        is_long: bool
+        content: List[str] = field(default_factory=list)
+
+    @dataclass
+    class JtblEntry:
+        count: int
+        base: int
+
     def __init__(self, config: Config) -> None:
         super().__init__(config)
+        self._jtbls: Dict[int, AsmProcessorSH2.JtblEntry] = {}
+        self._imms: Dict[int, AsmProcessorSH2.ImmEntry] = {}
+        self._relocs: Dict[int, str] = {}
+
+    def preprocess_objdump(self, objdump: str) -> str:
+        new_lines = []
+        lines = objdump.splitlines()
+
+        # Detect endianess, as there's no special marker for what is data
+        # and what is code (like in ARM) getting the endian is needed to
+        # correct literal pool bytes read as instructions
+        is_be = self._test_endian(lines)
```

Comment:
If you wanted to simplify this, we actually have the raw ELF available for parsing as well, from where the endian should be possible to read out from the header. Or (better?) you could do like "sh4" vs "sh4el", where `arch.big_endian` (accessible from `config.arch` in `__init__`) is statically known based on what the user passes in.